### PR TITLE
MVP First Version

### PR DESCRIPTION
**Pull request by kegor**
In this PR can be found back-end changes needed for following UI: https://github.com/groupdocs-total/GroupDocs.Total-Angular/pull/161.

**TODO:**
- [x] Add client folder with Metadata Angular UI npm-package usage.

### CODE REVIEW

**Review by Pavel-Teplitsky — APPROVED** [review]

**Review by Pavel-Teplitsky — APPROVED** [review]

File: src/client/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
+- [React](https://reactjs.org)
+  - `ng add @nrwl/react`
+- Web (no framework frontends)
+  - `ng add @nrwl/web`
+- [Nest](https://nestjs.com)
+  - `ng add @nrwl/nest`
+- [Express](https://expressjs.com)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 18
           

Complexity increasing per file
==============================
- src/Products/Common/Config/ConfigurationValuesGetter.cs  8
- src/Products/Common/Entity/Web/UploadedDocumentEntity.cs  1
- src/Products/Common/Config/ApplicationConfiguration.cs  5
- src/Products/Common/Entity/Web/FileDescriptionEntity.cs  1
- src/App_Start/FilterConfig.cs  1
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
- src/Products/Common/Config/CommonConfiguration.cs  1
- src/AppDomainGenerator/DomainGenerator.cs  2
- src/Products/Common/Entity/Web/LoadDocumentEntity.cs  1
- src/Products/Common/Entity/Web/ExceptionEntity.cs  1
- src/Products/Common/Config/GlobalConfiguration.cs  1
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
- src/Products/Common/Util/Comparator/FileNameComparator.cs  2
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/Controllers/MetadataController.cs  1
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/client/apps/metadata-e2e/src/plugins/index.js  1
- src/Products/Common/Util/Directory/DirectoryUtils.cs  4
- src/App_Start/RouteConfig.cs  1
- src/Products/Common/Util/Comparator/FileDateComparator.cs  2
- src/Products/Common/Entity/Web/PageDescriptionEntity.cs  1
- src/Products/Common/Resources/Resources.cs  5
- src/Products/Common/Util/Parser/YamlParser.cs  2
- src/Global.asax.cs  1
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- GroupDocs.Metadata.MVC.Test/MetadataControllerTest.cs  1
- src/Products/Common/Config/ServerConfiguration.cs  1
- src/Products/Common/Entity/Web/PostedDataEntity.cs  1
- src/App_Start/WebApiConfig.cs  1
         

Clones added
============
- src/Products/Metadata/Controllers/MetadataApiController.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988&bid=16279296)

File: src/client/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  - `ng add @nrwl/express`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+- [Node](https://nodejs.org)
+  - `ng add @nrwl/node`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+Nx supports many plugins which add capabilities for developing different types of applications and different tools.
+
+These capabilities include generating applications, libraries, .etc as well as the devtools to test, and build projects as well.
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
+- [React](https://reactjs.org)
+  - `ng add @nrwl/react`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+## Adding capabilities to your workspace
+
+Nx supports many plugins which add capabilities for developing different types of applications and different tools.
+
+These capabilities include generating applications, libraries, .etc as well as the devtools to test, and build projects as well.
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+Run `ng g @nrwl/angular:lib my-lib` to generate a library.
+
+> You can also use any of the plugins above to generate libraries as well.
+
+Libraries are sharable across libraries and applications. They can be imported from `@client/mylib`.
+
+## Development server
+
+Run `ng serve my-app` for a dev server. Navigate to http://localhost:4200/. The app will automatically reload if you change any of the source files.
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[no-literal-urls] Don’t use literal URLs without angle brackets](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+These capabilities include generating applications, libraries, .etc as well as the devtools to test, and build projects as well.
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
+- [React](https://reactjs.org)
+  - `ng add @nrwl/react`
+- Web (no framework frontends)
+  - `ng add @nrwl/web`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+- [Nest](https://nestjs.com)
+  - `ng add @nrwl/nest`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md (hunk 4 above)
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+Nx supports many plugins which add capabilities for developing different types of applications and different tools.
+
+These capabilities include generating applications, libraries, .etc as well as the devtools to test, and build projects as well.
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
+- [React](https://reactjs.org)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+## Adding capabilities to your workspace
+
+Nx supports many plugins which add capabilities for developing different types of applications and different tools.
+
+These capabilities include generating applications, libraries, .etc as well as the devtools to test, and build projects as well.
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md (hunk 2 above)
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+- [React](https://reactjs.org)
+  - `ng add @nrwl/react`
+- Web (no framework frontends)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+  - `ng add @nrwl/web`
+- [Nest](https://nestjs.com)
+  - `ng add @nrwl/nest`
+- [Express](https://expressjs.com)
+  - `ng add @nrwl/express`
+- [Node](https://nodejs.org)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+Below are some plugins which you can add to your workspace:
+
+- [Angular](https://angular.io)
+  - `ng add @nrwl/angular`
+- [React](https://reactjs.org)
+  - `ng add @nrwl/react`
+- Web (no framework frontends)
+  - `ng add @nrwl/web`
+- [Nest](https://nestjs.com)
```
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md (hunk 7 above)
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

File: src/client/README.md (hunk 8 above)
Comment (bobkovalex) [inline src/client/README.md]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988)

**Review by Pavel-Teplitsky — APPROVED** [review]

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Common/Util/Directory/DirectoryUtils.cs  4
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/Products/Common/Entity/Web/LoadDocumentEntity.cs  1
- src/Products/Common/Config/GlobalConfiguration.cs  1
- src/Products/Common/Util/Parser/YamlParser.cs  2
- GroupDocs.Metadata.MVC.Test/MetadataControllerTest.cs  1
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- src/Products/Common/Config/ServerConfiguration.cs  1
- src/App_Start/WebApiConfig.cs  1
- src/App_Start/FilterConfig.cs  1
- src/Products/Common/Util/Comparator/FileNameComparator.cs  2
- src/Products/Common/Util/Comparator/FileDateComparator.cs  2
- src/App_Start/RouteConfig.cs  1
- src/Products/Common/Entity/Web/PageDescriptionEntity.cs  1
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/Products/Common/Config/ApplicationConfiguration.cs  5
- src/Global.asax.cs  1
- src/Products/Common/Entity/Web/ExceptionEntity.cs  1
- src/Products/Common/Entity/Web/UploadedDocumentEntity.cs  1
- src/Products/Common/Config/CommonConfiguration.cs  1
- src/Controllers/MetadataController.cs  1
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/client/apps/metadata-e2e/src/plugins/index.js  1
- src/Products/Common/Entity/Web/PostedDataEntity.cs  1
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
- src/Products/Common/Config/ConfigurationValuesGetter.cs  8
- src/AppDomainGenerator/DomainGenerator.cs  2
- src/Products/Common/Entity/Web/FileDescriptionEntity.cs  1
- src/Products/Common/Resources/Resources.cs  5
         

Clones added
============
- src/Products/Metadata/Controllers/MetadataApiController.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988&bid=16279296)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Common/Entity/Web/UploadedDocumentEntity.cs  1
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/Products/Common/Config/CommonConfiguration.cs  1
- src/Products/Common/Entity/Web/PostedDataEntity.cs  1
- src/Products/Common/Entity/Web/FileDescriptionEntity.cs  1
- src/Products/Common/Resources/Resources.cs  5
- src/App_Start/FilterConfig.cs  1
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- src/Products/Common/Entity/Web/PageDescriptionEntity.cs  1
- src/Products/Common/Util/Parser/YamlParser.cs  2
- src/Products/Common/Config/GlobalConfiguration.cs  1
- src/Products/Common/Util/Directory/DirectoryUtils.cs  4
- src/Products/Common/Config/ApplicationConfiguration.cs  5
- src/App_Start/RouteConfig.cs  1
- src/App_Start/WebApiConfig.cs  1
- src/Products/Common/Config/ServerConfiguration.cs  1
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
- src/Products/Common/Entity/Web/ExceptionEntity.cs  1
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/Controllers/MetadataController.cs  1
- src/AppDomainGenerator/DomainGenerator.cs  2
- src/Products/Common/Util/Comparator/FileDateComparator.cs  2
- src/Products/Common/Config/ConfigurationValuesGetter.cs  8
- GroupDocs.Metadata.MVC.Test/MetadataControllerTest.cs  1
- src/Products/Common/Util/Comparator/FileNameComparator.cs  2
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
- src/client/apps/metadata-e2e/src/plugins/index.js  1
- src/Global.asax.cs  1
- src/Products/Common/Entity/Web/LoadDocumentEntity.cs  1
         

Clones added
============
- src/Products/Metadata/Controllers/MetadataApiController.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-metadata/GroupDocs.Metadata-for-.NET-MVC/pullRequest?prid=4918988&bid=16279296)

**Pull request merged**